x-pack/legacy/plugins/apm/server/lib/traces/get_trace_items.ts

```diff
@@ -16,27 +19,39 @@ import { Setup } from '../helpers/setup_request';
 
 export async function getTraceItems(traceId: string, setup: Setup) {
   const { start, end, client, config } = setup;
+  const maxTraceItems = config.get<number>('apm_oss.maxTraceItems');
 
   const params: SearchParams = {
     index: [
       config.get('apm_oss.spanIndices'),
       config.get('apm_oss.transactionIndices')
     ],
     body: {
-      size: 1000,
+      size: maxTraceItems,
       query: {
         bool: {
           filter: [
             { term: { [TRACE_ID]: traceId } },
             { terms: { [PROCESSOR_EVENT]: ['span', 'transaction'] } },
             { range: rangeFilter(start, end) }
-          ]
+          ],
+          should: {
+            exists: { field: PARENT_ID }
+          }
         }
-      }
+      },
+      sort: [
+        { _score: { order: 'asc' } },
+        { [TRANSACTION_DURATION]: { order: 'desc' } },
+        { [SPAN_DURATION]: { order: 'desc' } }
+      ]
     }
   };
 
   const resp = await client.search<Transaction | Span>(params);
 
-  return resp.hits.hits.map(hit => hit._source);
+  return {
+    items: resp.hits.hits.map(hit => hit._source),
+    exceedsMax: resp.hits.total > maxTraceItems
```

Comment:
Note: From version 7.0 elasticsearch will not produce accurate counts above 10,000 docs. We added a workaround for this this (`rest_total_hits_as_int`) so accurate counts are always returned but we'll remove that in https://github.com/elastic/kibana/issues/34915. We just have to remember to update this accordingly when we get around to that.